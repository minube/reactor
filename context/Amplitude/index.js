export { ConsumerAmplitude, ProviderAmplitude } from './Amplitude';
