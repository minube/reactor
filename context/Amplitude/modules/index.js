import queryString from './queryString';

export { queryString } // eslint-disable-line
