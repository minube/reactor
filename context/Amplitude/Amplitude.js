import { node, shape, string } from 'prop-types';
import React, { createContext, PureComponent } from 'react';
import { Platform } from 'react-native';

import { SHAPE } from '../../common';
import { queryString } from './modules';
import PKG from '../../package.json';

const IS_PRODUCTION = process.env.NODE_ENV === 'production';
const ENDPOINT = 'https://api.amplitude.com/httpapi';
const Context = createContext();
const { Provider, Consumer: ConsumerAmplitude } = Context;

class ProviderAmplitude extends PureComponent {
  _logEvent = ({ type, ...props } = {}) => {
    const { props: { key, session } } = this;

    if (!IS_PRODUCTION) console.log('⚡️ProviderAmplitude.LogEvent', type, props);

    if (Platform.OS === 'web') {
      const amplitude = require('amplitude-js').getInstance(); //eslint-disable-line
      amplitude.init(key);
      amplitude.logEvent(type, props);
      return true;
    }

    return fetch(ENDPOINT, {
      headers: { 'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8' },
      method: 'POST',
      body: queryString({
        api_key: key,
        event: JSON.stringify({
          ...session,
          event_type: type,
          event_properties: props,
          library: { name: PKG.name, version: PKG.version },
          platform: Platform.OS,
          timestamp: new Date().getTime(),
        }),
      }),
    });
  }

  render() {
    const { _logEvent, props: { children } } = this;

    return (
      <Provider value={{ logEvent: _logEvent }}>
        { children }
      </Provider>
    );
  }
}

ProviderAmplitude.propTypes = {
  children: node,
  key: string,
  session: shape(SHAPE.SESSION),
};

ProviderAmplitude.defaultProps = {
  children: undefined,
  key: IS_PRODUCTION ? '68fcc232a01a8a3490a4b59c9b819137' : '7c8dfb36a0efb5c6fe9d7e4940f3e3ae',
  session: undefined,
};

export { ConsumerAmplitude, ProviderAmplitude };
