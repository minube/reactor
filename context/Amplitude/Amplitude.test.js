import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';

import { ConsumerAmplitude, ProviderAmplitude } from './Amplitude';

global.fetch = () => {};

describe('<ProviderAmplitude>', () => {
  it('renders', () => {
    const tree = TestRenderer.create(<ProviderAmplitude />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('dispatch _logEvent', () => {
    const renderer = TestRenderer.create(<ProviderAmplitude  />);

    renderer.getInstance()._logEvent('hello');
  });

  it('renders in DOM environment', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ProviderAmplitude />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

