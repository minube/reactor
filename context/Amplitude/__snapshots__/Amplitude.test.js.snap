// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProviderAmplitude> renders 1`] = `null`;
